### PR TITLE
refactor: remove dataclassy from managers

### PR DESCRIPTION
**Pull request by evanblank3**
### What I did
 Removes `dataclassy` from all the managers and makes them regular python classes. Uses `injected_before_use` class to act as a property placeholder for the injected properties

### How I did it
Adds `injected_before_use` class 
 Adds `__init__()` methods to pass the required non-injected properties by keyword.

### How to verify it
All tests run

### Checklist

- [ ] Passes all linting checks (pre-commit and CI jobs)
- [ ] New test cases have been added and are passing
- [ ] Documentation has been updated
- [ ] PR title follows [Conventional Commit](https://www.conventionalcommits.org/en/v1.0.0/) standard (will be automatically included in the changelog)

### CODE REVIEW

**Review by unparalleled-js — DISMISSED** [review]
Looks good as a first step to deleting `dataclassy`.

Next step is converting the API classes to pydantic models.

**Comment by evanblank3** [comment]
After discussion with @fubuloubu moving back to the Class level injection of properties:
- only have to set them once
- setting them in __init__() doesn't make them class vars anymore
- fault detection, it raises if it's not set and you try to access it

**Comment by unparalleled-js** [comment]
> After discussion with @fubuloubu moving back to the Class level injection of properties:
> 
> * only have to set them once
> * setting them in **init**() doesn't make them class vars anymore
> * fault detection, it raises if it's not set and you try to access it

I don't really understand these benefits!

* You only have to set them once in `init` as well?
* What is the benefit of them being class vars? Since we are passing in references to the init anyway
* It also raises when you don't properly initialize the manager with one?

File: src/ape/__init__.py [hunk 1]
```diff
@@ -22,52 +22,65 @@
...
+_ProjectManager.compilers = compilers
+_ProjectManager.networks = networks
+_ProjectManager.converter = _converters
 Project = _partial(
     _ProjectManager,
-    config=config,
-    compilers=compilers,
-    networks=networks,
-    converter=_converters,
 )
```
Comment (fubuloubu) [inline src/ape/__init__.py:79]: let's remove this partial! just change the import to `from .managers.project import ProjectManager as Project`

File: src/ape/__init__.py [hunk 2]
```diff
@@ -22,52 +22,65 @@
...
 """User-facing class for instantiating Projects (in addition to the currently
 active ``project``). See :class:`ape.managers.project.ProjectManager`."""
 
-project = Project(config.PROJECT_FOLDER)
+project = Project(path=config.PROJECT_FOLDER)
 """The currently active project. See :class:`ape.managers.project.ProjectManager`."""
 
 Contract = _partial(_Contract, networks=networks, converters=_converters)
```
Comment (fubuloubu) [inline src/ape/__init__.py:86]: I'd like to get rid of this partial too, by turning `_Contract` into an object, or somehow just use `ContractInstance` (but I don't think that'd work quite as well.

File: src/ape/managers/chain.py [hunk 3]
```diff
@@ -195,12 +194,9 @@ class AccountHistory(_ConnectedChain):
 
     _map: Dict[AddressType, List[ReceiptAPI]] = {}
 
-    # TODO: Inject ConversionManager into AccountHistory
     @cached_property
     def _convert(self) -> Callable:
-        from ape import convert
-
-        return convert
+        return self._converters.convert
```
Comment (fubuloubu) [inline src/ape/managers/chain.py:199]: Nice

File: src/ape/managers/chain.py [hunk 4]
```diff
@@ -314,7 +312,9 @@ def account_history(self) -> AccountHistory:
         A mapping of transactions from the active session to the account responsible.
         """
         if self.chain_id not in self._account_history_map:
-            history = AccountHistory(self._networks)  # type: ignore
+            AccountHistory._networks = self._networks
+            AccountHistory._converters = self._converters
```
Comment (fubuloubu) [inline src/ape/managers/chain.py:316]: Should actually be able to do this injection inside of `__init__`! So you don't have to think about it later.

File: src/ape/managers/config.py [hunk 5]
```diff
@@ -47,9 +44,20 @@ class ConfigManager:
     contracts_folder: Path = None  # type: ignore
     dependencies: Dict[str, str] = {}
     deployments: Dict[str, Dict[str, List[DeploymentConfig]]] = {}
-    plugin_manager: PluginManager
+    plugin_manager: ClassVar[PluginManager] = injected_before_use()  # type: ignore
     _plugin_configs_by_project: Dict[str, Dict[str, ConfigItem]] = {}
 
+    def __init__(
+        self,
+        *,
```
Comment (fubuloubu) [inline src/ape/managers/config.py:52]: Why is this here?

**Comment by fubuloubu** [comment]
> * You only have to set them once in `init` as well?

For some of the classes that are not singletons, you would have to do this every time you init. It gets annoying. Class vars are set once and forget!

> * What is the benefit of them being class vars? Since we are passing in references to the init anyway

The managers are singletons, so you don't notice the benefit as much, but here is a good example of how it helps: https://github.com/ApeWorX/ape/pull/435#discussion_r793780770

> * It also raises when you don't properly initialize the manager with one?

This is good fault handling that you forgot something

File: src/ape/managers/config.py (hunk 5 above)
Comment (evanblank3) [reply src/ape/managers/config.py:52]: The * ensures that all params passed after it are done by keyword and not position.

Comment (fubuloubu) [reply src/ape/managers/config.py:52]: ah okay. but won't this ignore any args provided?

File: src/ape/__init__.py (hunk 1 above)
Comment (fubuloubu) [reply src/ape/__init__.py:79]: Also, here is one example where we made ugly code cleaner w/ dependency injection!

File: src/ape/managers/config.py (hunk 5 above)
Comment (evanblank3) [reply src/ape/managers/config.py:52]: It will get very angry about the args, do we want to support *arg, **kwargs on these init methods?

File: src/ape/__init__.py (hunk 2 above)
Comment (evanblank3) [reply src/ape/__init__.py:83]: I'll need to discuss the requirements/workflow around this change, whenever you have a moment.

File: src/ape/managers/config.py (hunk 5 above)
Comment (fubuloubu) [reply src/ape/managers/config.py:52]: No, I would just remove it if that's the case

File: src/ape/__init__.py (hunk 2 above)
Comment (evanblank3) [reply src/ape/__init__.py:83]: Opening a ticket specific to this refactor work, removing partial here and moving _Contract creation inside ContractContainer

**Review by unparalleled-js — APPROVED** [review]
LGTM
